### PR TITLE
ENH: stats.bootstrap: update warning to mention np.min

### PR DESCRIPTION
**Pull request by jcampbell**
Improves warning message for edge cases, based on potentially incremental strategy identified in gh-15883.

### CODE REVIEW

**Review by tupui — CHANGES_REQUESTED** [review]
Thank you @jcampbell. There are CI failure which relate to this PR. We are matching this warning message in some tests. Hence they have to be updated.

**Comment by jcampbell** [comment]
Thank you -- updating!

File: scipy/stats/_resampling.py [hunk 1]
```diff
@@ -87,8 +87,9 @@ def _percentile_along_axis(theta_hat_b, alpha):
     for indices, alpha_i in np.ndenumerate(alpha):
         if np.isnan(alpha_i):
             # e.g. when bootstrap distribution has only one unique element
-            msg = ("The bootstrap distribution is degenerate; the "
-                   "confidence interval is not defined.")
+            msg = ("The BCa confidence interval cannot be calculated. This problem "
+                   "is known to occur when the distribution is degenerate or the "
+                   "statistic is np.min.")
```
[the remark this answers is not in the document]
Comment (jcampbell) [reply scipy/stats/_resampling.py]: Thank you -- updated in same commit with the test message change.

File: scipy/stats/_resampling.py [hunk 2]
```diff
@@ -87,8 +87,11 @@ def _percentile_along_axis(theta_hat_b, alpha):
     for indices, alpha_i in np.ndenumerate(alpha):
         if np.isnan(alpha_i):
             # e.g. when bootstrap distribution has only one unique element
-            msg = ("The bootstrap distribution is degenerate; the "
-                   "confidence interval is not defined.")
+            msg = (
+                "The BCa confidence interval cannot be calculated."
+                " This problem is known to occur when the distribution is degenerate or the"
+                " statistic is np.min."
+            )
```
Comment (tupui) [inline scipy/stats/_resampling.py:94]: Should fix the linter
```suggestion
            msg = (
                "The BCa confidence interval cannot be calculated."
                " This problem is known to occur when the distribution"
                " is degenerate or the statistic is np.min."
            )
```

**Review by tupui — COMMENTED** [review]
Is there a reason why you did not go with @mdhaber's suggestion?

> The observed value of the statistic is less than or equal to all values in the bootstrap distribution. This is known to occur when, for example, the data is degenerate or the statistic is np.min. The BCa confidence interval cannot be calculated.

This seems more complete and nuanced (example).

**Comment by jcampbell** [comment]
I actually modified the suggestion because I was concerned that it is *too* specific. For example, if our sample includes `nan`, we will encounter the same problem when bootstrapping the `mean` statistic, but not for the *exact* reason specified. Since this change is already designed to clarify that there are other situations that can cause this problem, I thought it would be best to lead with the net result and then give examples of why that could happen.

**Comment by tupui** [comment]
I will let @mdhaber weight in.

**Review by mdhaber — APPROVED** [review]
Yup, this is fine. I agree that it can happen in other cases, and we were too specific before, so let's not make the same mistake. Looks like CI is happy with the changes in this PR, too. Anything else, @tupui?

**Review by tupui — APPROVED** [review]
All good then. Thank you @jcampbell and @mdhaber for the review.

**Pull request merged**